lib/rules/no-disallowed-lwc-imports.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+const LWC_SUPPORTED_APIS = new Set([
+    // Common APIs
+    'LightningElement',
+    'api',
+    'track',
+    'wire',
+
+    // From "@lwc/engine-core"
+    'getComponentDef',
+    'getComponentConstructor',
+    'isComponentConstructor',
+    'createContextProvider',
+    'readonly',
+    'register',
+    'setFeatureFlagForTest',
+    'unwrap',
+
+    // From "@lwc/engine-dom"
+    'hydrateComponent',
+    'buildCustomElementConstructor',
+    'createElement',
+
+    // From "@lwc/engine-server"
+    'renderComponent',
+]);
```

Comment:
This is based on [our current list](https://github.com/salesforce/lwc/blob/647de31f30be1084253a85b8b850fb72de7294ac/packages/%40lwc/babel-plugin-component/src/constants.js#L37-L56).